### ISSUE TITLE
SELECT on tuple relations are broken for mixed ASC/DESC clustering order

### ISSUE THREAD

**Issue by tgrabiec** [scylladb/scylla#2050]
Mirror of https://issues.apache.org/jira/browse/CASSANDRA-7281

```
cqlsh:blues> create table foo (a int, b int, c int, PRIMARY KEY (a, b, c)) WITH CLUSTERING ORDER BY (b DESC, c ASC);
cqlsh:blues> INSERT INTO foo (a, b, c) VALUES (0, 2, 0);
cqlsh:blues> INSERT INTO foo (a, b, c) VALUES (0, 1, 0);
cqlsh:blues> INSERT INTO foo (a, b, c) VALUES (0, 1, 1);
cqlsh:blues> INSERT INTO foo (a, b, c) VALUES (0, 0, 0);
cqlsh:blues> SELECT * FROM foo WHERE a=0;

 a | b | c
---+---+---
 0 | 2 | 0
 0 | 1 | 0
 0 | 1 | 1
 0 | 0 | 0

(4 rows)
```

```
cqlsh:blues> SELECT * FROM foo WHERE a=0 AND (b, c) > (1, 0);
```

Actual:
```
 a | b | c
---+---+---
 0 | 1 | 1
 0 | 0 | 0

(2 rows)

```

Expected:
```

 a | b | c
---+---+---
 0 | 2 | 0
 0 | 1 | 1
```

**Comment by slivne** [comment]
@eliransin can you please have a look at the cql_additional_tests.py - it has a few (~10) tests marked as requiring this issue to be fixed - yet when trying to enable all of them and testing 3.0.rc1 - some of them still fail.

**Comment by eliransin** [comment]
@slivne I'll have a look. 
10x.

**Comment by eliransin** [comment]
@slivne I have updated the tests, it is indeed another issue.

**Comment by eliransin** [comment]
Closing since it passes unit testing and dtests are blocked on another issue.
The resolving commit is [83628f5](https://github.com/scylladb/scylla/commit/83628f5881b85abd74a405375512fec1a0be99a9)
